src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -476,6 +478,14 @@ private async Task RestoreAsync(bool forceRestore, RestoreOperationSource restor
                             // Display the restore opt out message if it has not been shown yet
                             await l.WriteHeaderAsync();
 
+                            // initialize PackageExtractionContext to allow logging inside the restore call
+                            _nuGetProjectContext.PackageExtractionContext = new PackageExtractionContext(
```

Comment:
This still might not fix it when package is restored through restore bar either on NuGet Manager UI or PMC. 

Scenario - create a packages.config based project, install a package, close UI. Delete packages folder and open Manager UI. It will show a yellow bar indicating missing packages and option to restore (same happens when opened PMC). Click restore which should also log any warning.

May be the right place to update existing code https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.PackageManagement/IDE/PackageRestoreManager.cs#L292 to have the right logger?